test/tests.py

```diff
@@ -829,25 +829,31 @@ def test(self):
                 raise TestFail(f'cannot load profile data')
         else:
             raise TestFail(f'stats file not generated')
+
+
 class DumpInputFilesTestCase(CTestCase):
     '''Test that the DMLC_DUMP_INPUT_FILES variable works.  It creates a
     tarball of input DML files that can be compiled standalone.'''
-    __slots__ = ()
+    __slots__ = ('prefix',)
+    def __init__(self, *args, prefix='', **kwargs):
+        self.prefix = prefix
+        super().__init__(*args, **kwargs)
     def test(self):
         super().test()
         dir = Path(self.scratchdir) / 'dumped'
         dir.mkdir()
         with tarfile.open(Path(self.scratchdir) / f'T_{self.shortname}.tar.xz',
                           'r:xz') as tf:
             tf.extractall(dir)
-        if is_windows():
-            assert (dir / '_' / os.path.basename(self.filename)).is_file()
-        else:
+        assert (dir.joinpath(*self.prefix)
+                / os.path.basename(self.filename)).is_file()
+        if not is_windows():
             # This does not work on Windows, for unknown reasons.
             # Seems related to symlink semantics somehow, but no
             # need to explore deeper until we have a use case for it.
-            cmd = dmlc + ['_/' + os.path.basename(self.filename),
-                          self.shortname]
+            cmd = dmlc + [
+                join(*self.prefix + [os.path.basename(self.filename)]),
```

Comment:
Sooo the precedence rules will parse this as `*(self.prefix + [os.path.basename(self.filename)])`?
That's... not very intuitive, even though I guess it makes sense. Python's weird.

Reply:
True, I now added a parenthesis and that made it easier to read.